build.sh

```diff
@@ -1,3052 +1,129 @@
-
-: ${REALM_COCOA_PLATFORMS:="osx ios watchos tvos"}
-: ${REALM_DOTNET_COCOA_PLATFORMS:="ios-no-bitcode"}
-
-ANDROID_DIR="android-lib"
-ANDROID_PLATFORMS="arm arm-v7a arm64 mips x86 x86_64"
-
-NODE_DIR="node-lib"
-
-CONFIG_VERSION=1
-CURRENT_PLATFORM="win"
-if [ "`uname`" = "Darwin" ]; then
-  CURRENT_PLATFORM="osx"
-fi
-if [ "`uname`" = "Linux" ]; then
-  CURRENT_PLATFORM="linux"
-fi
-
-usage()
-{
-    cat 1>&2 << EOF
-Unspecified or bad mode '$MODE'.
-Available modes are:
-    config:
-    clean:
-    build:
-    build-m32:                          build in 32-bit mode
-    build-arm-benchmark:
-    build-config-progs:
-    build-osx:
-    build-ios:
-    build-ios-no-bitcode:
-    build-watchos:
-    build-tvos:
-    build-android:
-    build-cocoa:
-    build-dotnet-cocoa:
-    build-osx-framework:
-    build-node:
-    build-node-package:
-    test:
-    test-debug:
-    check:
-    check-debug:
-    memcheck:
-    memcheck-debug:
-    check-testcase:
-    check-testcase-debug:
-    memcheck-testcase:
-    memcheck-testcase-debug:
-    asan:
-    asan-debug:
-    build-test-ios-app:                 build an iOS app for testing core on device
-    test-ios-app:                       execute the core tests on device
-    leak-test-ios-app:                  execute the core tests on device, monitor for leaks
-    gdb:
-    gdb-debug:
-    gdb-testcase:
-    gdb-testcase-debug:
-    performance:
-    benchmark:
-    benchmark-*:
-    lcov:
-    gcovr:
-    show-install:
-    release-notes-prerelease:
-    release-notes-postrelease:
-    get-version:
-    set-version:
-    copy-tools:
-    install:
-    install-prod:
-    install-devel:
-    uninstall:
-    uninstall-prod:
-    uninstall-devel:
-    test-installed:
-    wipe-installed:
-    src-dist:
-    bin-dist:
-    dist-config:
-    dist-clean:
-    dist-build:
-    dist-build-iphone:
-    dist-test:
-    dist-test-debug:
-    dist-install:
-    dist-uninstall:
-    dist-test-installed:
-    dist-status:
-    dist-pull:
-    dist-checkout:
-    dist-copy:
-    jenkins-pull-request:               Run by Jenkins for each pull request whenever it changes
-    jenkins-pipeline-unit-tests:        Run by Jenkins as part of the core pipeline whenever master changes
-    jenkins-pipeline-coverage:          Run by Jenkins as part of the core pipeline whenever master changes
-    jenkins-pipeline-address-sanitizer: Run by Jenkins as part of the core pipeline whenever master changes
-    jenkins-valgrind:
-EOF
-}
-
-map_ext_name_to_dir()
-{
-    local ext_name
-    ext_name="$1"
-    case $ext_name in
-        *) echo "realm_$ext_name";;
+#!/bin/bash
+
+#Set Script Name variable
+SCRIPT=$(basename "${BASH_SOURCE[0]}")
+DIR=$(dirname "${BASH_SOURCE[0]}")
+
+# Set fonts
+NORM="\e[0m"
+BOLD="\e[1m"
+RED="${BOLD}\e[31m"
+
+# Number of cores
+CORES=$(getconf _NPROCESSORS_ONLN)
+
+function usage {
+    echo "${BOLD}Usage:${NORM} ${SCRIPT} -t <build_type> -o <target_os> -v <version> [-a <android_abi>]"
```

Comment:
The version of `echo` that ships with OS X doesn't support `\e` escape sequences. That results in this help output being mangled. `printf` does the right thing.